app/src/main/java/com/orange/ouds/app/ui/components/checkbox/CheckboxItemDemoState.kt

```diff
@@ -0,0 +1,112 @@
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.mapSaver
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.state.ToggleableState
+import com.orange.ouds.app.R
+
+@Composable
+fun rememberCheckboxItemDemoState(
+    toggleableState: ToggleableState = ToggleableState.Off,
+    icon: Boolean = false,
+    divider: Boolean = true,
+    inverted: Boolean = false,
+    enabled: Boolean = true,
+    readOnly: Boolean = false,
+    error: Boolean = false,
+    text: String = stringResource(id = R.string.app_components_common_text_label),
+    helperText: String? = null
+) = rememberSaveable(toggleableState, icon, divider, inverted, enabled, readOnly, error, text, helperText, saver = CheckboxItemDemoState.Saver) {
+    CheckboxItemDemoState(toggleableState, icon, divider, inverted, enabled, readOnly, error, text, helperText)
+}
+
+class CheckboxItemDemoState(
+    toggleableState: ToggleableState,
+    icon: Boolean,
+    divider: Boolean,
+    inverted: Boolean,
+    enabled: Boolean,
+    readOnly: Boolean,
+    error: Boolean,
+    text: String,
+    helperText: String?
+) {
+    companion object {
+        val Saver = run {
+            val toggleableStateKey = "toggleableState"
+            val iconKey = "icon"
+            val dividerKey = "divider"
+            val invertedKey = "inverted"
+            val enabledKey = "enabled"
+            val readOnlyKey = "readOnly"
+            val errorKey = "error"
+            val textKey = "text"
+            val helperTextKey = "helperText"
+            mapSaver(
+                save = { state ->
+                    mapOf(
+                        toggleableStateKey to state.toggleableState,
+                        iconKey to state.icon,
+                        dividerKey to state.divider,
+                        invertedKey to state.inverted,
+                        enabledKey to state.enabled,
+                        readOnlyKey to state.readOnly,
+                        errorKey to state.error,
+                        textKey to state.text,
+                        helperTextKey to state.helperText
+                    )
+                },
+                restore = { map ->
+                    CheckboxItemDemoState(
+                        map[toggleableStateKey] as ToggleableState,
+                        map[iconKey] as Boolean,
+                        map[dividerKey] as Boolean,
+                        map[invertedKey] as Boolean,
+                        map[enabledKey] as Boolean,
+                        map[readOnlyKey] as Boolean,
+                        map[errorKey] as Boolean,
+                        map[textKey] as String,
+                        map[helperTextKey] as String
```

Comment:
This should be a `String?` instead of `String` (the app crashes when the device rotates to landscape).